matrix_benchmarking/common.py

```diff
@@ -40,18 +58,31 @@ def settings_to_key(settings):
         return "|".join(f"{k}={settings[k]}" for k in sorted(settings) if k != "stats")
 
     @staticmethod
-    def all_records(settings, setting_lists):
+    def all_records(settings, setting_lists, include_local=True, include_lts=False) -> Iterator[MatrixEntry]:
         for settings_values in sorted(itertools.product(*setting_lists)):
             settings.update(dict(settings_values))
             key = Matrix.settings_to_key(settings)
+
             try:
-                yield Matrix.processed_map[key]
+                entry = Matrix.processed_map[key]
             except KeyError:
                 continue # missing experiment
+            if (entry.is_lts and include_lts) or (not entry.is_lts and include_local):
+                yield entry
```

Comment:
I wonder if one way or another, `all_records` should return the local entries first 🤔 
(see https://github.com/openshift-psap/ci-artifacts/pull/697 for why it's important)

actually, I think the sorting should be done here, at the core level.

but I'm not sure what's the best way to expose it, and what should be the default 🤔 

with the current code:
```python
for settings_values in sorted(itertools.product(*setting_lists)):
```
the order is defined by `setting_list`, which itself comes from there:
```python
def do_plot(self, ordered_vars, settings, setting_lists, variables, cfg):
```
so in the web UI (that you're not using, AFAIK), you can change the order to decide what comes first.

I guess I would prefer having another flag `local_first`, which behaves this way:
```python
def all_records(settings, setting_lists, include_local=True, include_lts=False, local_first=True) -> Iterator[MatrixEntry]:
    if local_first:
        if not (include_local and include_lts):
            raise ValueError("local_first=True cannot be set without include_local=True and include_lts=True")
        yield from all_records(settings, setting_list, include_local=True, include_lts=False, local_first=False)
        yield from all_records(settings, setting_list, include_local=False, include_lts=True, local_first=False)
        return
```